activestorage/test/models/blob_test.rb

```diff
@@ -43,6 +43,21 @@ class UserWithHasOneAttachedDependentFalse < User
     assert_equal "text/plain", blob.content_type
   end
 
+  test "create after upload extracts content_type from io when no content_type given" do
+    blob = create_csv_blob content_type: nil
+    assert_equal "text/csv", blob.content_type
+  end
+
+  test "create after upload extracts content_type when extract_content_type_from_io: true" do
+    blob = create_csv_blob data: "Article,dates,analysis\n1, 2, 3", content_type: "text/csv", extract_content_type_from_io: true
+    assert_equal "message/news", blob.content_type
+  end
```

Comment:
The new option defaults to true, so this test is superfluous. The preexisting tests above cover this behavior.

Reply:
Removed this test in this commit 539e880fe4731ef430a857a35ec9efc167edf263